src/charm.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python3
+
+import logging
+import subprocess
+from pathlib import Path
+
+from ops.charm import CharmBase
+from ops.main import main
+from ops.model import ActiveStatus
+
+
+log = logging.getLogger(__name__)
+
+
+class UbuntuCharm(CharmBase):
+    def __init__(self, *args):
+        super().__init__(*args)
+
+        self.framework.observe(self.on.config_changed, self.on_config_changed)
+
+        if self.unit.is_leader():
+            self.set_version()
+
+        self.unit.status = ActiveStatus()
+
+    def set_version(self):
+        try:
+            for line in Path("/etc/lsb-release").read_text().splitlines():
+                k, v = line.strip().split("=")
+                if k.strip() == "DISTRIB_RELEASE":
+                    self.unit.set_workload_version(v.strip())
+        except Exception:
+            log.exception("Error getting release")
+
+    def on_config_changed(self, event):
```

Comment:
On the theme of what might end up getting used as a pattern elsewhere I know there is some preference for private methods here unless they're intended to be consumed by other charms. We've also had feedback asking for docstrings for all methods.